sherpa/utils/__init__.py

```diff
@@ -3716,7 +3716,52 @@ def __call__(self, a, b, c):
 
 
 def bisection(fcn, xa, xb, fa=None, fb=None, args=(), maxfev=48, tol=1.0e-6):
+    """A basic root finding algorithm that uses standard bisection
 
+    Bisection is a relatively slow method for root finding, but it guaranteed to
+    work for a continuous function with a root in a bracketed interval; in other
+    words the function must undergo a sign change between the bracketing values.
+
+    See https://en.wikipedia.org/wiki/Bisection_method for a description of the
+    bisection method.
+
+    Parameters
+    ----------
+    fcn : callable
+        The function with a root. The function signature is ``fcn(x, *args)``.
+    xa : float
+        Lower limit of the bracketing interval
+    xb : float
+        Upper limit of the bracketing interval
+    fa : float or None
+        Function value at ``xa``. This parameter is optional and can be passed
+        to save time in cases where ``fcn(xa, *args)`` is already known and
+        function evaluation takes a long time. If `None`, it will be
+        calculated.
+    fb : float or None
+        Function value at ``xb``. This parameter is optional and can be passed
+        to save time in cases where ``fcn(xb, *args)`` is already known and
+        function evaluation takes a long time. If `None`, it will be
+        calculated.
+    args : tuple
+        Additional parameters that will be passed through to ``fcn``.
+    maxfev : int
+        Maximal number of function evaluations
+    tol : float
+        The root finding algorithm stops if a value x with
+        ``abs(fcn(x)) < tol`` is found.
+
+    Returns
+    -------
+    out : list
```

Comment:
If I were reviewing this as new code I'd argue that we return a dataclass here, since we have a number of routines that return the same thing and it not a particularly-friendly return type! However, I'm not suggesting we change the code (at least today ;-)